packages/pluggableWidgets/datagrid-dropdown-filter-web/src/components/__tests__/FilterComponent.spec.tsx

```diff
@@ -0,0 +1,212 @@
+    { caption: "3", value: "_3" }
+];
+
+describe("Filter selector", () => {
+    describe("with single selection", () => {
+        describe("renders correctly", () => {
+            it("with options", () => {
+                const component = shallow(<FilterComponent options={defaultOptions} filterDispatcher={jest.fn()} />);
+
+                expect(component).toMatchSnapshot();
+            });
+            it("with no options", () => {
+                const component = shallow(<FilterComponent options={[]} filterDispatcher={jest.fn()} />);
+
+                expect(component).toMatchSnapshot();
+            });
+            it("with ariaLabel", () => {
+                const component = shallow(
+                    <FilterComponent options={defaultOptions} ariaLabel="my label" filterDispatcher={jest.fn()} />
+                );
+
+                expect(component).toMatchSnapshot();
+            });
+            it("with emptyOptioncaption", () => {
+                const component = shallow(
+                    <FilterComponent
+                        options={defaultOptions}
+                        emptyOptionCaption={"find me"}
+                        filterDispatcher={jest.fn()}
+                    />
+                );
+
+                expect(component).toMatchSnapshot();
+                expect(component.find("input").first().prop("value")).toBe("find me");
+            });
+        });
+        it("selects default option", () => {
+            const defaultOption = defaultOptions[0];
+            const component = shallow(
+                <FilterComponent
+                    options={defaultOptions}
+                    filterDispatcher={jest.fn()}
+                    defaultValue={defaultOption.value}
+                />
+            );
+
+            const input = component.find("input").first();
+
+            expect(input.prop("value")).toBe(defaultOption.caption);
+        });
+
+        describe("when value changes", () => {
+            it("calls filterDispatcher when value changes", () => {
+                const filterDispatcher = jest.fn();
+                const component = shallow(
+                    <FilterComponent options={defaultOptions} filterDispatcher={filterDispatcher} />
+                );
+
+                const item = component.find("li").first();
+                item.simulate("click");
+
+                expect(filterDispatcher).toBeCalled();
+            });
+            it("shows selected option on input value", () => {
+                const defaultOption = defaultOptions[1];
+                const component = shallow(
+                    <FilterComponent
+                        options={defaultOptions}
+                        filterDispatcher={jest.fn()}
+                        defaultValue={defaultOption.value}
+                    />
+                );
+                expect(component.find("input").first().prop("value")).toBe(defaultOption.caption);
```

Comment:
Assuming you want `component.find("input").first().prop("value")` to be a var: it cannot be. Because input value gets updated after the click but not reflected if we extract this on a var

Reply:
Apologies, I was rather referring to `component.find("input").first()`

Reply:
Still holds true :/